icu4c/source/common/uloc_tag.cpp

```diff
@@ -31,17 +31,17 @@ typedef struct VariantListEntry {
 } VariantListEntry;
 
 /* struct holding a single attribute value */
-typedef struct AttributeListEntry {
+struct AttributeListEntry : public icu::UMemory {
     const char              *attribute;
```

Comment:
Optional: Change all of these char* fields into CharString, and avoid the strPool?

Reply:
One step at a time, changing the data type in the data structure would also require changes in the code that works on these structures.